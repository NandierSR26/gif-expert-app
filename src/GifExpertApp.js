import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

/* componente principal de la app */
export const GifExpertApp = () => {

    // almacena el nombre de las categorias | guarda el nombre de una nueva cetegoria
    const [categories, setCategories] = useState(['Naruto']);

    // const handleAdd = () => {
    //     setCategories([...categories, 'Naruto']);
    //     // setCategories( cat => [ ...cat, 'Naruto' ]);
    // }

    return (
        <>
            <h2>App de GIfs</h2>
            <AddCategory setCategories={setCategories} />
            <hr></hr>
            <ul>
                {
                    categories.map((category) => (
                        <GifGrid
                            key={ category }
                            category={ category }
                        />
                    ))
                }
            </ul>
        </>
    )
}
