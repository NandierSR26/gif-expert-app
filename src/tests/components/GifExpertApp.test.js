import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../../GifExpertApp';


describe('pruebas de GifExpertApp', () => {

    test('mostrar el componente', () => {
        const wapper = shallow( <GifExpertApp />);

        expect( wapper ).toMatchSnapshot();
    });

    test('debe de mostrar una lista de categorias', () => {

        const categories = ['one punch', 'naruto'];
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } />);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    });
    
    
});
