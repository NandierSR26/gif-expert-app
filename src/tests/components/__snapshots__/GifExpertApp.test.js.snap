// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas de GifExpertApp debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="one punch"
      key="one punch"
    />
    <GifGrid
      category="naruto"
      key="naruto"
    />
  </ul>
</Fragment>
`;

exports[`pruebas de GifExpertApp mostrar el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul />
</Fragment>
`;
